### PR TITLE
H3 integration

### PR DESCRIPTION
**Pull request by BHYCHIK**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- New Feature

Short description (up to few sentences):

Intergate H3 function geoToH3 to clickhouse.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Looks like a draft: no tests, the function implemented only for constant argument.

**Comment by BHYCHIK** [comment]
> Looks like a draft: no tests, the function implemented only for constant argument.

Thanks for review. It was done for internal needs. I will update MR with tests and non-constants arguments.

**Comment by BHYCHIK** [comment]
> Looks like a draft: no tests, the function implemented only for constant argument.

I have added support of non-constants arguments, tests and russian documentation. Could you review it once again, please?

**Comment by alexey-milovidov** [comment]
You need to fix builds. You can find the relevant info in "Details" link:
https://clickhouse-builds.s3.yandex.net/4724/32c338556c9db19faba6e3a9c7edf7d3b26ce03a/report.html

File: docs/ru/query_language/functions/geo.md [hunk 1]
```diff
@@ -88,6 +88,37 @@ pointInPolygon((x, y), [(a, b), (c, d) ...], ...)
...
+│   1 │
+└─────┘
+```
+
+## geoToH3
+
+Получает H3 индекс точки (lat, lon) с заданным разрешением
+
+```
+pointInPolygon(lat, lon, resolution)
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/geo.md]: Wrong function name in example.

File: docs/ru/query_language/functions/geo.md [hunk 2]
```diff
@@ -88,6 +88,37 @@ pointInPolygon((x, y), [(a, b), (c, d) ...], ...)
...
+```
+
+**Входные значения**
+
+- `lat` - географическая широта. Тип данных — [Float64](../../data_types/float64.md).
+- `lon` - географическая долгота. Тип данных — [Float64](../../data_types/float64.md).
+- функция поддерживает также многоугольники с дырками (вырезанными кусками). Для этого случая, добавьте многоугольники, описывающие вырезанные куски, дополнительными аргументами функции. Функция не поддерживает неодносвязные многоугольники.
```
Comment (alexey-milovidov) [inline docs/ru/query_language/functions/geo.md]: Wrong text. BTW, you don't have to write russian docs if you don't know russian. English is enough.

**Comment by alexey-milovidov** [comment]
?

**Comment by alexey-milovidov** [comment]
?

**Pull request merged**